backend/providers/serpapi_common.py

```
@@ -1,10 +1,66 @@
+        try:
+            resp = await client.get(SERPAPI_ENDPOINT, params=q)
+        except httpx.HTTPError as exc:
+            last_err = f"request failed: {exc}"
+            continue
+        if resp.status_code == 401:
+            last_err = "401 (invalid key)"
+            serp_keys.advance()
+            continue
+        try:
+            data = resp.json()
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)